### ISSUE TITLE
Training with TOOD , loss_cls remains unchanged,loss_bbox has negtive value.

### ISSUE THREAD

**Issue by dsp6414** [open-mmlab/mmdetection#7952]
Config:
dataset_type = 'yyminstDataset'
data_root = '/home/chen/OD/mmdet_tutorial/datasets/yymnist'
img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', with_bbox=True),
    dict(type='Resize', img_scale=(1333, 800), keep_ratio=True),
    dict(type='RandomFlip', flip_ratio=0.5),
    dict(
        type='Normalize',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        to_rgb=True),
    dict(type='Pad', size_divisor=32),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels'])
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=(1333, 800),
        flip=False,
        transforms=[
            dict(type='Resize', keep_ratio=True),
            dict(type='RandomFlip'),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='Pad', size_divisor=32),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img'])
        ])
]
data = dict(
    samples_per_gpu=2,
    workers_per_gpu=2,
    train=dict(
        type='yyminstDataset',
        ann_file='/home/chen/OD/mmdet_tutorial/datasets/yymnist/train.txt',
        img_prefix='/home/chen/OD/mmdet_tutorial/datasets/yymnist/images/',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='LoadAnnotations', with_bbox=True),
            dict(type='Resize', img_scale=(1333, 800), keep_ratio=True),
            dict(type='RandomFlip', flip_ratio=0.5),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='Pad', size_divisor=32),
            dict(type='DefaultFormatBundle'),
            dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels'])
        ],
        classes=('0', '1', '2', '3', '4', '5', '6', '7', '8', '9')),
    val=dict(
        type='yyminstDataset',
        ann_file='/home/chen/OD/mmdet_tutorial/datasets/yymnist/train.txt',
        img_prefix='/home/chen/OD/mmdet_tutorial/datasets/yymnist/images/',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(1333, 800),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(
                        type='Normalize',
                        mean=[123.675, 116.28, 103.53],
                        std=[58.395, 57.12, 57.375],
                        to_rgb=True),
                    dict(type='Pad', size_divisor=32),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ],
        classes=('0', '1', '2', '3', '4', '5', '6', '7', '8', '9')),
    test=dict(
        type='yyminstDataset',
        ann_file='/home/chen/OD/mmdet_tutorial/datasets/yymnist/train.txt',
        img_prefix='/home/chen/OD/mmdet_tutorial/datasets/yymnist/images/',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(1333, 800),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(
                        type='Normalize',
                        mean=[123.675, 116.28, 103.53],
                        std=[58.395, 57.12, 57.375],
                        to_rgb=True),
                    dict(type='Pad', size_divisor=32),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ],
        classes=('0', '1', '2', '3', '4', '5', '6', '7', '8', '9')))
evaluation = dict(interval=5, metric='mAP')
optimizer = dict(type='SGD', lr=0.0025, momentum=0.9, weight_decay=0.0001)
optimizer_config = dict(grad_clip=None)
lr_config = dict(
    policy='step',
    warmup=None,
    warmup_iters=500,
    warmup_ratio=0.001,
    step=[8, 11])
runner = dict(type='EpochBasedRunner', max_epochs=100)
checkpoint_config = dict(interval=10)
log_config = dict(interval=5, hooks=[dict(type='TextLoggerHook')])
custom_hooks = [dict(type='SetEpochInfoHook')]
dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = None
resume_from = None
workflow = [('train', 1)]
model = dict(
    type='TOOD',
    backbone=dict(
        type='ResNet',
        depth=50,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        frozen_stages=1,
        norm_cfg=dict(type='BN', requires_grad=True),
        norm_eval=True,
        style='pytorch',
        init_cfg=dict(type='Pretrained', checkpoint='torchvision://resnet50')),
    neck=dict(
        type='FPN',
        in_channels=[256, 512, 1024, 2048],
        out_channels=256,
        start_level=1,
        add_extra_convs='on_output',
        num_outs=5),
    bbox_head=dict(
        type='TOODHead',
        num_classes=10,
        in_channels=256,
        stacked_convs=6,
        feat_channels=256,
        anchor_type='anchor_free',
        anchor_generator=dict(
            type='AnchorGenerator',
            ratios=[1.0],
            octave_base_scale=8,
            scales_per_octave=1,
            strides=[8, 16, 32, 64, 128]),
        bbox_coder=dict(
            type='DeltaXYWHBBoxCoder',
            target_means=[0.0, 0.0, 0.0, 0.0],
            target_stds=[0.1, 0.1, 0.2, 0.2]),
        initial_loss_cls=dict(
            type='FocalLoss',
            use_sigmoid=True,
            activated=True,
            gamma=2.0,
            alpha=0.25,
            loss_weight=1.0),
        loss_cls=dict(
            type='QualityFocalLoss',
            use_sigmoid=True,
            activated=True,
            beta=2.0,
            loss_weight=1.0),
        loss_bbox=dict(type='GIoULoss', loss_weight=2.0)),
    train_cfg=dict(
        initial_epoch=4,
        initial_assigner=dict(type='ATSSAssigner', topk=9),
        assigner=dict(type='TaskAlignedAssigner', topk=13),
        alpha=1,
        beta=6,
        allowed_border=-1,
        pos_weight=-1,
        debug=False),
    test_cfg=dict(
        nms_pre=1000,
        min_bbox_size=0,
        score_thr=0.05,
        nms=dict(type='nms', iou_threshold=0.6),
        max_per_img=100))
work_dir = './tutorial_exps'
seed = 0
gpu_ids = range(0, 1)

training result:
2022-05-10 14:05:40,214 - mmdet - INFO - Epoch [1][965/2394]    lr: 2.500e-03, eta: 18:58:12, time: 0.287, data_time: 0.014, memory: 2465, loss_cls: 25.0000, loss_bbox: -2.4469, loss: 22.5531
2022-05-10 14:05:41,675 - mmdet - INFO - Epoch [1][970/2394]    lr: 2.500e-03, eta: 18:58:17, time: 0.291, data_time: 0.014, memory: 2465, loss_cls: 25.0000, loss_bbox: -7.3118, loss: 17.6882
2022-05-10 14:05:43,189 - mmdet - INFO - Epoch [1][975/2394]    lr: 2.500e-03, eta: 18:58:37, time: 0.304, data_time: 0.014, memory: 2465, loss_cls: 25.0000, loss_bbox: -2.5183, loss: 22.4817
2022-05-10 14:05:44,662 - mmdet - INFO - Epoch [1][980/2394]    lr: 2.500e-03, eta: 18:58:43, time: 0.293, data_time: 0.014, memory: 2465, loss_cls: 25.0000, loss_bbox: -2.6166, loss: 22.3834
2022-05-10 14:05:46,152 - mmdet - INFO - Epoch [1][985/2394]    lr: 2.500e-03, eta: 18:58:57, time: 0.299, data_time: 0.015, memory: 2465, loss_cls: 25.0000, loss_bbox: 0.5863, loss: 25.5863
2022-05-10 14:05:47,585 - mmdet - INFO - Epoch [1][990/2394]    lr: 2.500e-03, eta: 18:58:55, time: 0.286, data_time: 0.013, memory: 2465, loss_cls: 25.0000, loss_bbox: -6.3538, loss: 18.6462
2022-05-10 14:05:49,022 - mmdet - INFO - Epoch [1][995/2394]    lr: 2.500e-03, eta: 18:58:55, time: 0.288, data_time: 0.014, memory: 2465, loss_cls: 25.0000, loss_bbox: -0.4833, loss: 24.516
